src/utils/coinUtils.js

```diff
@@ -282,6 +282,14 @@ export const parseTransaction = (
     output.scriptHash = reverseBufferToHex(hash256Sync(output.script.toRaw()))
   }) && bcoinTx
 
+export const parseJsonTransaction = (
+  txJson: Object,
+  bcoinTx: any = primitives.MTX.fromJSON(txJson)
+) =>
+  !txJson.inputs.forEach(({ coin, prevout }) => {
```

Comment:
This is confusing. I'm not sure what this returns. A boolean or an object..? It's supposed to return a bcoinTx but I'm not sure how it would. Please deconstruct to make this easier to read.

Reply:
Sorry bout that. I guess I'm so used to 'code golfing' that for me this seems like simple legitimate syntax :)
I rewrote the function in a more "human" readable way.

Reply:
I understand the syntax now but it doesn't seem to match the original code. Specifially the `!txJson.inputs` and the `&& bcoinTx` part. I'll save that to another convo for a later time. Approved